### DISCUSSION TITLE
Get parent note title for backlink on new file creation from empty link

### DISCUSSION THREAD

**Discussion by cedardocster** [SilentVoid13/Templater#799] [Help] [answered]
# How to get parent note title for backlink on new file creation from empty link

I know how to add the *current* note title (i.e. the newly formed note) by using this in the template:

'# <% tp.file.title%>

but how do I get the title of the note I linked *from*?  I.e. the parent note title in order to create a hard back link?

E.g. "Note A":
Hi there. Here's a [[new file]]

"new file":
Back link: [[Note A]]
'# new file

I'm (relatively)  new to Obsidian (and GitHub - sorry about originally submitting in wrong spot), and I imagine I may have to do some sort of JavaScript in my new file template to get the name of the most recently modified file (which I think would be the parent file), but is there an easier built-in way to get the parent note name for when you click or tap on a link to a file that doesn't exist yet?

**Comment by shabegom** [comment]
Not an easy problem to solve. Can you provide some more context around when you would run the script? Are you creating the new file with `tp.file.create_new()` or using the templater command?

**Comment by welpdx** [comment, marked as answer]
One easy implementation what you want is this:
```js
<%* 
let files = this.app.workspace.getLastOpenFiles();
let lastFile = this.app.vault.getAbstractFileByPath(files[0]); // t file
tR+= `you last file is: [[` + lastFile.basename +`]]`
%>
```

Add it to Folder Templates to trigger it when you create a new note or you can just use it as is. 

<details>
           <summary style="font-size:20">Demo</summary>
           <img src="https://i.imgur.com/j1QbtXP.gif" alt="demo" />
</details>

**Reply by cedardocster** [reply]
Yes! That's it!! Thank you so much :)

**Comment by sisyphus-prime** [comment]
My use case is to automatically insert the parent URL as a property into all newly created notes. For me, the parent URL is the note I am in when I create the new note.  

Trying the original solution was getting funky results because it uses the last open variable as opposed to the last active file (which could be different). It appears the following command produces the desired output. I have not rigorously tested this solution, and I am new to Obsidian and scripting so user beware. 

> parent-url:` "[[<% this.app.workspace.getActiveFile().basename %>]]"
